src/snaphu/_util.py

```diff
@@ -1,190 +1,206 @@
 from __future__ import annotations
 
-import itertools
+import io
 import os
 import shutil
-from collections.abc import Callable, Generator, Iterable, Iterator
+from collections.abc import Callable, Generator, Iterator
 from contextlib import contextmanager
-from dataclasses import dataclass
 from pathlib import Path
-from tempfile import mkdtemp
+from tempfile import mkdtemp, mkstemp
 
 import numpy as np
-from numpy.typing import ArrayLike
+from numpy.typing import ArrayLike, DTypeLike
 
 from .io import InputDataset, OutputDataset
 
 __all__ = [
-    "BlockIterator",
-    "ceil_divide",
-    "copy_blockwise",
     "nan_to_zero",
+    "new_unique_file",
+    "read_from_file",
     "scratch_directory",
+    "slices",
+    "write_to_file",
 ]
 
 
-def as_tuple_of_int(ints: int | Iterable[int]) -> tuple[int, ...]:
+def new_unique_file(
+    *,
+    dir_: str | os.PathLike[str] | None = None,
+    prefix: str | None = None,
+    suffix: str | None = None,
+) -> Path:
     """
-    Convert the input to a tuple of ints.
+    Create a new file with a unique file name.
 
     Parameters
     ----------
-    ints : int or iterable of int
-        One or more integers.
+    dir_ : str or path-like or None, optional
+        The directory that the file will be created in. If None, the default temporary
+        directory is chosen as though by ``tempfile.gettempdir()``. Defaults to None.
+    prefix : str or None, optional
+        An optional file name prefix. If None, a default prefix is used. Defaults to
+        None.
+    suffix : str or None, optional
+        An optional file name prefix. If None, there will be no suffix. Defaults to
+        None.
 
     Returns
     -------
-    out : tuple of int
-        Tuple containing the input(s).
+    pathlib.Path
+        The absolute path to created file.
     """
-    try:
-        return (int(ints),)  # type: ignore[arg-type]
-    except TypeError:
-        return tuple([int(i) for i in ints])  # type: ignore[union-attr]
+    if dir_ is not None:
+        dir_ = os.fspath(dir_)
+    file, filename = mkstemp(dir=dir_, prefix=prefix, suffix=suffix)
+    os.close(file)
```

Comment:
This isn't an issue for us because we're not using `mkstemp` for security reasons in an adversarial environment. 

We're just using `mkstemp` to ensure that any new scratch files are guaranteed to have unique names so that multiple snaphu-py processes running in parallel with the same scratch directory don't stomp on each other. That property is still guaranteed even if we close the file descriptor after creating the file, so long as all snaphu-py processes always use `mkstemp`.

Note that, prior to this PR, we were already ignoring the returned file descriptor and already re-opening the file in a subprocess.

Of course, if a user deletes or overwrites some scratch files created by snaphu-py while it's running, then we'd have a problem. But there's no way for me to protect against that kind of malicious misuse, even if I keep the file descriptor open.